File: src/server/gkill/api/req_res/submit_kftl_text_response.go
package req_res

import "github.com/mt3hr/gkill/src/server/gkill/api/message"

type SubmitKFTLTextResponse struct {
	Messages []*message.GkillMessage `json:"messages"`
	Errors   []*message.GkillError   `json:"errors"`

	// Created は確定したレコード。KFTLは1つのテキストから複数のKyouを作るのに、
	// 2026-08-24 まで応答は「記録しました」の1文だけで、件数も種別もIDも返らなかった。
	// 確定は1つの SQLite トランザクション（commit_tx と同じ）なので、失敗したときは何も
	// 残らず空。冪等キーで再送を畳んだときも実行していないので空。
	Created []*SubmitKFTLTextCreated `json:"created"`
}

// SubmitKFTLTextCreated はKFTLが書いた1件。
type SubmitKFTLTextCreated struct {
	ID       string `json:"id"`
	DataType string `json:"data_type"`
	// Updated は新規作成ではなく既存レコードの更新であることを表す（打刻の終了）。
	Updated bool `json:"updated"`
}

File: src/client/classes/api/req_res/submit-kftl-text-response.ts
'use strict'

import { GkillAPIResponse } from '../gkill-api-response'

// サーバの KFTL が実際に書いた1件（Go 側 req_res.SubmitKFTLTextCreated）。
export interface SubmitKFTLTextCreated {
    id: string
    data_type: string
    // 新規作成ではなく既存レコードの更新（打刻の終了）
    updated: boolean
    // 書いた記録の関連時刻（打刻の終了は終了時刻）。JSON なので文字列で届く
    related_time: string
}

export class SubmitKFTLTextResponse extends GkillAPIResponse {

    // 実際に書いた記録。失敗したときは何も残らず空（commit は1つの SQLite トランザクション。ADR-0219）。
    // 冪等キーで再送を畳んだときも実行していないので空
    created: Array<SubmitKFTLTextCreated> | null = null

    constructor() {
        super()
    }

}
